python/tvm/rpc/server.py

```diff
@@ -119,6 +119,15 @@ def download_linked_module(file_name):
     return temp
 
 
+def _serve_loop(sock, load_library, work_path=None):
+    """Server loop"""
+    sockfd = sock.fileno()
+    temp = _server_env(load_library, work_path)
+    _ffi_api.ServerLoop(sockfd)
+    if not work_path:
+        temp.remove()
```

Comment:
curious, why need the temp? it seems this function only have one caller, and it will give the work_path.

Reply:
The answer to most of your comments is that I reverted part of (https://github.com/apache/tvm/pull/15187) so that _server_loop can be pickled. The code here comes from the code before that commit, and I didn't simplify it. I assumed there was a reason for its style, but I'll take another pass at this.

Reply:
Thanks for your explanation, I think just move the closure out of the function _serving will fix the issue, so from my side, it is better to keep the code simple, so please keep it as same as possible with the current implementation if there isn't reasonable requirement, thanks.